Simperium/src/main/java/com/simperium/client/Channel.java

```diff
@@ -1455,6 +1470,10 @@ private void processRemoteChanges()
                     if (!remoteChange.isError() && remoteChange.isRemoveOperation()) {
                         dequeueLocalChangesForKey(remoteChange.getKey());
                     }
+
+                    if (!remoteChange.isError()) {
+                        mBucket.notifyOnSyncObjectListeners(remoteChange.getKey());
+                    }
```

Comment:
```suggestion
                    if (!remoteChange.isError()) {
                        mBucket.notifyOnSyncObjectListeners(remoteChange.getKey());

                        if (remoteChange.isRemoveOperation()) {
                            dequeueLocalChangesForKey(remoteChange.getKey());
                        }                        
                    }
```

Would it make sense to merge these two `if` statements?  Do you think the nested `if`s would be less readable?

Reply:
I thought about it but I felt that the nesting would be more confusing since the notification and the dequeueing are independent things. if you don't like having both of them we could create new flags indicating the type of change

```java
bool needToRemove = !remoteChange.isError() && removeChange.isRemoveOperation();
bool needToNotify = !removeChange.isError();

if (needToRemove) {
	dequeueLocalChangesForKey(remoteChange.getKey());
}

if (needToNotify) {
	mBucket.notifyOnSyncObjectListeners(remoteChange.getKey());
}
```

Split into names it reads even more as independent things to me; what are your thoughts?

Reply:
I'm fine with either really.  The flags may make the purpose a little clearer, but I don't think they're necessary.  I'll make this conversation as resolved.